src/device/mod.rs

```diff
@@ -1124,9 +1124,16 @@ impl Runtime {
             if self.features.nicknames {
                 peers = self.requested_state.collect_dns_nickname_records();
             }
-
             peers.extend(self.requested_state.collect_dns_records());
 
+            // Insert wildcard for subdomains
+            let wildcarded_peers: Records = peers
+                .clone()
+                .into_iter()
+                .map(|(name, ip)| (format!("*.{}", name), ip))
+                .collect();
```

Comment:
It might not be that performance critical, but I think we could reduce the number of allocations slightly this way:

```suggestion
            let wildcarded_peers: Records = peers
                .iter()
                .map(|(name, ip)| (format!("*.{}", name), ip.clone()))
                .collect();
```

Reply:
And if I'm not mistaken, `HashMap::extend` takes an iterator, so the `collect` might be unnecessary here

Reply:
Nice, thanks for pointing that out. It still needs to collect() though, otherwise compiler refuses to extend `peers` as the iterator borrowed peers already.